components/script/canvas_state.rs

```diff
@@ -60,6 +60,39 @@ use crate::dom::paintworkletglobalscope::PaintWorkletGlobalScope;
 use crate::dom::textmetrics::TextMetrics;
 use crate::script_runtime::CanGc;
 
+// Firefox limits width/height to 32767 pixels and Chromium to 65535 pixels,
+// but slows down dramatically before it reaches that limit.
+// We limit by area instead, giving us larger maximum dimensions,
+// in exchange for a smaller maximum canvas size.
+const MAX_CANVAS_AREA: u64 = 32768 * 8192;
+// Max width/height to 65535 in CSS pixels.
+const MAX_CANVAS_SIZE: u64 = 65535;
+
+#[derive(Clone, Debug, JSTraceable, MallocSizeOf)]
+/// Helps observe states of canvas draw target.
```

Comment:
Please put documentation above the `derive` line and perhaps expand to:

```suggestion
/// Information about the canvas draw target size and whether the size is valid.
```